src/components/visualizations/chart/chartOptionsBuilder.ts

```diff
@@ -1117,16 +1111,47 @@ function getCategories(
     return [];
 }
 
-export function getCategoriesForTwoAttributes(
-    viewByAttribute: IUnwrappedAttributeHeadersWithItems,
-    viewByTwoAttributes: IViewByTwoAttributes
-) {
-    const insideAttributes  = viewByAttribute.items.reduce(getDistinctAttributeHeaderName, []);
-    const outsideAttributes = viewByTwoAttributes.items.reduce(getDistinctAttributeHeaderName, []);
+/**
+ * Transform
+ *      viewByTwoAttributes: {
+ *          parent: [P1, P1, P2, P2, P3],
+ *          children: [C1, C2, C1, C2, C2]
+ *      }
+ * to
+ *      [{
+ *          name: P1,
+ *          categories: [C1, C2]
+ *       }, {
+ *          name: P2,
+ *          categories: [C1, C2]
+ *       }, {
+ *          name: P3,
+ *          categories: [C2]
+ *       }]
+ * @param viewByTwoAttributes
+ */
+export function getCategoriesForTwoAttributes(viewByTwoAttributes: IViewByTwoAttributes): Array<{
+    name: string,
+    categories: string[]
+}> {
+    const { parent = [], children = [] } = viewByTwoAttributes;
+    const combinedResult = zip(parent, children).reduce(
```

Comment:
I don't think its necessary to use `zip` there.
you can simply use the third parameter in `reduce` to get the mapping item in `children` array.

Reply:
Fixed :)